src/commands/project.lisp

```diff
@@ -221,3 +224,85 @@
                                        (incf count))))
       (delete-buffers (list-project-buffers)))
     (message "~a buffers deleted." count)))
+
+
+
+(defvar *projects-history*)
+
+(defun history ()
+  "Return or create the projects' history struct.
+  The history file is saved on (lem-home)/history/projects"
+  (unless (boundp '*projects-history*)
+    (let* ((pathname (merge-pathnames "history/projects" (lem-home)))
+           (history (lem/common/history:make-history :pathname pathname)))
+      (setf *projects-history* history)))
+  *projects-history*)
+
+(defun remember-project (input)
+  "Add this project path to the history file.
+  Return t if the project was added, nil if it already existed."
+  (let* ((history (history))
+         (data (lem/common/history:history-data history))
+         ;; project roots (pathnames) are converted to strings for the
+         ;; string completion prompt.
+         (input (namestring input)))
+    (cond
+      ((not (find input data :test #'equal))
+       (lem/common/history:add-history history input)
+       (lem/common/history:save-file history)
+       input)
+      (:otherwise
+       nil))))
+
+(defun forget-project (input)
+  "Remove this project (string) from the projects' history file.
+  Return the input."
+  (let ((history (history)))
+    (lem/common/history:remove-history history input)
+    (lem/common/history:save-file history)
+    input))
+
+(defun saved-projects ()
```

Comment:
To facilitate encapsulation, why not define a function in history.lisp that returns the history as a list?

Reply:
You are right, I'll try to facilitate encapsulation. Thanks for the review.